From c13ce3a5ec779ada94bfbf558bebd166f525cb10 Mon Sep 17 00:00:00 2001
From: Franck LENORMAND <franck.lenormand@nxp.com>
Date: Wed, 13 Jun 2018 16:40:07 +0200
Subject: [PATCH 4199/5242] MLK-18082: crypto: caam: fix index when setting
 ctrlpriv->jr[]

commit  90e4e9cc67a3e6f9d12e36669a86409a39e55274 from
https://source.codeaurora.org/external/imx/linux-imx.git

In case index == 00001000, the resulting index used to access
ctrlpriv->jr[] was 15 instead of the expected value of 0.

Signed-off-by: Franck LENORMAND <franck.lenormand@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/crypto/caam/ctrl.c |    2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/crypto/caam/ctrl.c b/drivers/crypto/caam/ctrl.c
index 864dc6d..4ae20aa 100644
--- a/drivers/crypto/caam/ctrl.c
+++ b/drivers/crypto/caam/ctrl.c
@@ -460,7 +460,7 @@ static int enable_jobrings(struct caam_drv_private *ctrlpriv, int block_offset)
 			/* Get actual job ring index from its offset
 			 * ex: CAAM JR2 offset 0x30000 index = 2
 			 */
-			while (index > 16)
+			while (index >= 16)
 				index = index >> 4;
 			index -= 1;
 			ctrlpriv->jr[index] = (struct caam_job_ring __force *)
-- 
1.7.9.5

